lib/helpers/model/discriminator.js

```diff
@@ -44,10 +44,8 @@ function validateDiscriminatorSchemasForEncryption(parentSchema, childSchema) {
     }
   }
 
-  function* allNestedPaths(schema) {
-    const { paths, singleNestedPaths } = schema;
-    yield* Object.keys(paths);
-    yield* Object.keys(singleNestedPaths);
+  function allNestedPaths(schema) {
```

Comment:
Just a follow up from https://github.com/Automattic/mongoose/pull/15320#discussion_r2044581203, which merged before I could make the change.